### PR TITLE
Expand the Spare5 Python API client

### PR DESCRIPTION
**Pull request by Jaidan**

### CODE REVIEW

**Comment by Jaidan** [comment]
@philipkimmey I think this is good enough for our purposes.  I won't know for sure until I start writing the command, but you should start looking through this before I get too deep into the command

File: spare5/resources/base.py [hunk 1]
```diff
@@ -1,18 +1,53 @@
...
 class ListResource(object):
+    _result = None
+
     def __init__(self, client):
         self.client = client
 
     def __iter__(self):
-        raise NotImplementedError()
+        if not hasattr(self, '_resource_class'):
+            raise NotImplementedError
```
Comment (philipkimmey) [inline spare5/resources/base.py]: raise an instance

File: spare5/resources/jobs.py [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
+    @property
+    def responses(self):
+        return Responses(self.client, self)
+
 
 class Jobs(ListResource):
     REQUIRED_PARAMS = ('num_responders',)
+    _resource_class = Job
+
+    def __init__(self, client, batch, result=None):
```
Comment (philipkimmey) [inline spare5/resources/jobs.py]: It's weird that the sig here doesn't match that of the parent class - I don't understand why if it was intentional.

**Comment by philipkimmey** [comment]
Couple small issues commented inline - otherwise lgtm :+1:

**Pull request merged**